File: 0686-repeated-string-match/0686-repeated-string-match.py
class Solution:
    def repeatedStringMatch(self, a: str, b: str) -> int:
        if(a == b):
            return 1
        if(b.count(a) < 0):
            return -1
        elif(a.count(b)>0):
            return 1
        s = a
        count = 1
        while(len(s) <= 10000):
            s += a
            count += 1
            if(s.count(b)>0):
                return count
        
        return -1
            
        
                
        

File: 0686-repeated-string-match/0686-repeated-string-match.cpp
class Solution {
public:
    int repeatedStringMatch(string a, string b) {
        if(a == b)return 1;
        string s = a;
        int index = b.find(a);
        
        int ind = a.find(b);
        if(ind >= 0)return 1;
        else{
            int count = 1;
            while(s.length() <= 10000){
                s += a;
                count++;
                ind = s.find(b);
                if(ind >=0 )return count;
            }
        }
        
        return -1;
    }
};